### ISSUE TITLE
Feature Request - More controls over Bars or User Roles

### ISSUE THREAD

**Issue by zhdenny** [karlomikus/bar-assistant#211]
This is a big request related to the User Roles and Bar Access Controls.

Can there be more access controls related to bars than just a simple on/off switch for users? Currently, the user role "General" is capable of adding cocktails and ingredients in ALL BARS they can see. This seems bad. It would be nice if Bars themselves had user-specific restrictions. For example, Allow an option for users to make changes to their own bars but not to other people's bars. There really is no user role to accomplish this.....
- The Guest role cannot change anything, only view
- The General role can add contents across everyone's bars.

I'm not sure what the best way to implement this is......maybe modifying user roles or maybe making more granular privilege controls for each bar created....and the bar creator gets to decide on those permissions.

Or at a bare minimum can there be a user role that only allows the user to only modify contents of Bars which they created. Because currently the user roles are not very useful to me if the General User can change contents of all the bars on the BarAssistant instance....and the Guest can only View.

Thanks! Great update! I love this app man!

Edit: I just realized that the General role cannot Edit/Delete cocktails or ingredients they do not own but they can ADD cocktails or ingredients into anyone's bar they can see. This is what I'd like more control over

**Comment by karlomikus** [comment]
I don't think I see a problem with this, is there any specific problem you are trying to solve?

- If you want a user to add his own cocktails/ingredients to a bar you assign a General role to him.
- If you want a user to only view things in a bar you assign a Guest role to him.

**Comment by zhdenny** [comment]
My Problem:

In my instance I have my Bar I created. I want to be the only one who can Add Cocktails/Ingredients to my bar.
In my instance I also have created various user accounts for different people. I want each of them to be able to do the following:
- View My Bar. Not modify it in anyway
- Create their own Bar and manage (add/remove/edit) their own Cocktails and Ingredients.

I don't see a way to do this currently. Currently, the additional accounts I created for users are capable of adding cocktails/ingredients into MY Bar. They are not able to EDIT anything in my bar but they can create cocktails/ingredients in it.
